src/threading.c

```diff
@@ -423,6 +423,7 @@ JL_DLLEXPORT jl_gcframe_t **jl_adopt_thread(void)
     JL_GC_PROMISE_ROOTED(ct);
     uv_random(NULL, NULL, &ct->rngState, sizeof(ct->rngState), 0, NULL);
     jl_atomic_fetch_add(&jl_gc_disable_counter, -1);
+    ct->world_age = jl_get_world_counter(); // root_task sets world_age to 1
```

Comment:
This seems fundamentally very wrong, as it mutates the world value and then returns, which is not particularly legal. Setting it to `1` generally legal, though I believe setting it to `0` is generally preferred?

Reply:
The issue is that setting it to this value means that the dispatch we do after this always fails. And this thread has never run any Julia code so it doesn’t really have any world. And ccallables just check the world anyway.
I guess it could be a invoke in world style

Reply:
Eh, sure, I guess that does make sense. Some code used to check for world==0 and assume that meant Julia wasn't initialized to be permitted to run on that thread, so anything non-zero is indeed probably better now. Still strange though to capture the counter value of when this thread was created